api/src/main/scala/com/lightbend/lagom/internal/api/ServiceReader.scala

```diff
@@ -160,7 +204,24 @@ object ServiceReader {
   }
 }
 
-case class SelfDescribingServiceCallStub[Id, Request, Response](method: Method, idType: Type, requestType: Type, responseType: Type) extends SelfDescribingServiceCall[Id, Request, Response] {
-  def invoke(id: Id, request: Request) = throw new NotImplementedError("This service call is a stub that should not be invoked")
-  def methodName = method.getName
-}
+/**
+ * Internal API.
+ *
+ * Service call holder that holds the original method reference that was passed to the descriptor when the service
+ * descriptor was defined.
+ *
+ * @param methodReference The method reference (a lambda object).
+ */
+private[lagom] case class MethodRefServiceCallHolder(methodReference: Any) extends ServiceCallHolder
+
+/**
+ * Internal API.
+ *
+ * Service call holder that's used by the service router and client implementor to essentially the raw id to and from
+ * invocations of the service call method.
+ */
+private[lagom] trait MethodServiceCallHolder extends ServiceCallHolder {
+  val method: Method
```

Comment:
It may be necessary to reflect over the method, which `MethodHandle` I don't think allows.